### ISSUE TITLE
Support for ALTER TABLE adding or altering columns NOT NULL

### ISSUE THREAD

**Issue by aarashy** [duckdb/duckdb#3248]
Hello!

I'm inquiring about support for the following ALTER TABLE statements:
```
D alter table foo add column f2 bigint not null;
Error: Parser Error: Adding columns with constraints not yet supported

D alter table foo alter column f2 set not null;
Error: Not implemented Error: ALTER TABLE option not supported yet!

D alter table foo alter column f2 drop not null;
Error: Not implemented Error: ALTER TABLE option not supported yet!
```

I saw that support for similar statements such as SET/DROP DEFAULT was made available in https://github.com/duckdb/duckdb/pull/612, and the foreign key work happening in https://github.com/duckdb/duckdb/pull/3195 seems related, but I haven't seen any other issues or activity about adding/setting non-null columns. 

I can work around the lack of support for these statements, but I was wondering to what extent the dev team considers supporting these a priority for upcoming releases.

**Comment by Mytherin** [comment]
This is one of our oldest issues (#57). I would say it is not a major priority currently, but happy to review a PR.

**Comment by gmiliarakis** [comment]
excuse me but how can DuckDB still not support ALTER TABLE ADD CONSTRAINT/ ADD column REFERENCES statements? How am I supposed to add a relation to an existing table? Drop the table and create a new one? I can't if it's referenced in other tables. Sample code:
`ALTER TABLE sequencing_runs ADD IF NOT EXISTS experiment_ids VARCHAR REFERENCES sequencing_experiments(id);`

`Parser Error: Adding columns with constraints not yet supported`

**Comment by hatamiarash7** [comment]
Many years later and not yet supported =))

**Comment by davetapley** [comment]
@Mytherin I'm confused: this was closed by #4185, but docs (and testing) imply it is not supported?

> In certain cases, constraints can also be added to a table using the [ALTER TABLE statement](https://duckdb.org/docs/lts/sql/statements/alter_table.html), but this is not currently supported for all constraints.

https://duckdb.org/docs/lts/sql/constraints

> ADD CONSTRAINT and DROP CONSTRAINT clauses are not yet supported in DuckDB.

https://duckdb.org/docs/lts/sql/statements/alter_table#add--drop-constraint
